pandas/core/arrays/integer.py

```diff
@@ -293,6 +295,44 @@ def __array__(self, dtype=None):
         """
         return self._coerce_to_ndarray()
 
+    _HANDLED_TYPES = (np.ndarray, numbers.Number)
+
+    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
+
+        out = kwargs.get('out', ())
+
+        for x in inputs + out:
+            if not isinstance(x, self._HANDLED_TYPES + (IntegerArray,)):
+                return NotImplemented
+
+        # for binary ops, use our custom dunder methods
+        result = ops.maybe_dispatch_ufunc_to_dunder_op(
+            self, ufunc, method, *inputs, **kwargs)
+        if result is not None:
+            return result
+
+        if (method == '__call__'
+                and ufunc.signature is None
+                and ufunc.nout == 1):
+            # only supports IntegerArray for now
+            args = [a._data for a in inputs]
+            masks = [a._mask for a in inputs]
+            result = ufunc(*args, **kwargs)
+            mask = np.logical_or.reduce(masks)
+            if result.dtype.kind in ('i', 'u'):
+                return IntegerArray(result, mask)
+            else:
+                result[mask] = np.nan
+                return result
+
+        # fall back to array for other ufuncs
+        inputs = tuple(
+            np.array(x) if isinstance(x, type(self)) else x
+            for x in inputs
+        )
+        return np.array(self).__array_ufunc__(
```

Comment:
You probably want to call the public ufunc again here, e.g., `return getattr(ufunc, method)(*inputs, **kwargs)`. Otherwise you don't invoke the dispatching mechanism again.

See [NDArrayOperatorsMixin](https://docs.scipy.org/doc/numpy-1.15.1/reference/generated/numpy.lib.mixins.NDArrayOperatorsMixin.html) for an example implementation.